tokio/src/runtime/metrics/io.rs

```diff
@@ -0,0 +1,29 @@
+#![cfg_attr(not(feature = "net"), allow(dead_code))]
+
+use std::sync::atomic::{AtomicU64, Ordering::Relaxed};
+
+#[derive(Default)]
+pub(crate) struct IoDriverMetrics {
+    pub(super) fd_count: AtomicU64,
```

Comment:
Is it worth tracking registered & deregistered separately? Right now, if an FD is registered then immediately deregistered, this will show up as the counter not moving. What is the value in tracking outstanding FDs? If registered & deregistered ops are tracked separately, churn will be detectable.

Separately, do we want to track `fd_count` or punt that to `tracing`? I am not sure. Thoughts @hawkw

Reply:
Wouldn't this catch things like leaked fds etc? I think there is value in tracking this internally to the driver in addition to what ever tracing can bring?

Reply:
tracking register & deregister separately would be a super set. Total fds == registers - deregisters.